drawee/src/main/java/com/facebook/drawee/generic/GenericDraweeHierarchy.java

```diff
@@ -275,12 +275,14 @@ public void setImage(Drawable drawable, float progress, boolean immediate) {
 
   @Override
   public void setProgress(float progress, boolean immediate) {
-    mFadeDrawable.beginBatchMode();
-    setProgress(progress);
-    if (immediate) {
-      mFadeDrawable.finishTransitionImmediately();
+    if (mFadeDrawable.getDrawable(PROGRESS_BAR_IMAGE_INDEX) != null) {
```

Comment:
This is purely a nit but it might be best to do the same as in `setProgress(float)` where it does an early return if the drawable is null rather than surrounding the remainder of the method in the if block.